### ISSUE TITLE
Unable to link with either Clang or MSVC, missing symbols on Win32

### ISSUE THREAD

**Issue by obscurelyme** [gfx-rs/wgpu-native#158]
```
C:\PROGRA~1\LLVM\bin\CLANG_~1.EXE -fuse-ld=lld-link -nostartfiles -nostdlib -g -Xclang -gcodeview -O0 -D_DEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrtd -Xlinker /subsystem:console CMakeFiles/HelloWgpu.dir/main.cpp.obj -o HelloWgpu.exe -Xlinker /implib:HelloWgpu.lib -Xlinker /pdb:HelloWgpu.pdb -Xlinker /version:0.0   C:/Users/ME/Projects/wgpu-native/target/debug/wgpu_native.lib  -luserenv.lib  -lws2_32.lib  -lDwmapi.lib  -ldbghelp.lib  -ld3dcompiler.lib  -lD3D12.lib  -lD3D11.lib  -lDXGI.lib  -lsetupapi.lib  -lBcrypt.lib -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 -loldnames && cmd.exe /C "cd /D C:\Users\ME\Projects\hello-wgpu\build && C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -noprofile -executionpolicy Bypass -file C:/Users/ME/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/Users/ME/Projects/hello-wgpu/build/HelloWgpu.exe -installedDir C:/Users/ME/Projects/hello-wgpu/build/vcpkg_installed/x64-windows/debug/bin -OutVariable out""
```

I'm looking to build out a small "Hello, World" using wgpu, but I'm getting missing symbols during the linker step. I've followed the steps in the wiki to pull down and compile this library and I have both a debug and release target built on my system. I can see from the clang linker step that the wgpu_native.lib was added to the list of libraries to link with posted above.

`C:/Users/ME/Projects/wgpu-native/target/debug/wgpu_native.lib`

And my main.cpp that I wish to build is quite simple

```
#include <iostream>
#include <webgpu-headers/webgpu.h>
#include <wgpu.h>

#define WGPU_TARGET_MACOS 1
#define WGPU_TARGET_LINUX_X11 2
#define WGPU_TARGET_WINDOWS 3
#define WGPU_TARGET_LINUX_WAYLAND 4

void logCallback(WGPULogLevel level, const char* msg) {
  std::string level_str;
  switch (level) {
  case WGPULogLevel_Error:
    level_str = "Error";
    break;
  case WGPULogLevel_Warn:
    level_str = "Warn";
    break;
  case WGPULogLevel_Info:
    level_str = "Info";
    break;
  case WGPULogLevel_Debug:
    level_str = "Debug";
    break;
  case WGPULogLevel_Trace:
    level_str = "Trace";
    break;
  default:
    level_str = "Unknown Level";
  }
  printf("[%s] %s\n", level_str.c_str(), msg);
}

void initializeLog() {
  wgpuSetLogCallback(logCallback);
  wgpuSetLogLevel(WGPULogLevel_Warn);
}

int main(int, char**) {
    initializeLog();
    std::cout << "WGPU Version: " << wgpuGetVersion() << std::endl;
    return 0;
}
```

The exact error is 

`lld-link: error: undefined symbol: void __cdecl wgpuSetLogCallback(void (__cdecl *)(enum WGPULogLevel, char const *))`
`lld-link: error: undefined symbol: void __cdecl wgpuSetLogLevel(enum WGPULogLevel)`
`lld-link: error: undefined symbol: unsigned int __cdecl wgpuGetVersion(void)`

**Comment by sqaxomonophonen** [comment]
Does it work if you wrap the `wgpu.h` (not `webgpu.h`) `#include` in `extern "C"`? That is:
```c++
extern "C" {
#include <wgpu.h>
}
```
That fixes it on my end, so I've open pull request #159 which fixes it in the header itself.

**Comment by kvark** [comment]
Fixed by #159 . Thank you!
